### PR TITLE
docs/resources: add the uml definitions and diagrams

### PR DESCRIPTION
**Pull request by leyao-daily**
This PR is to deliver the diagram of our class and the related puml files to generate the diagram.

### CODE REVIEW

File: docs/resources/pumls/engines/tensorflowengine.puml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        - _device: str
+        - _target: str
+        - _output_layer: list
+        - _drawing: dict
+        + path: str
+        + dtype: str
+        + device: str
+        + target: str
+        + output_layer: list
+        + drawing: dict
```
Comment (Hulongyin) [inline docs/resources/pumls/engines/tensorflowengine.puml]: (property)

File: docs/resources/pumls/engines/tensorflowengine.puml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    class TensorFlowPreprocessor {
```
Comment (Hulongyin) [inline docs/resources/pumls/engines/tensorflowengine.puml:23]: Add `__init__(input_size: Tuple[int, int], dtype: str)` function

File: docs/resources/pumls/engines/tensorflowengine.puml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        - _input_size: Tuple[int, int]
+        - _dtype: str
+        + preprocess(frame: np.ndarray): np.ndarray
+    }
+
+    class TensorFlowPostprocessor {
```
Comment (Hulongyin) [inline docs/resources/pumls/engines/tensorflowengine.puml:31]: Add `__init__(target: str, drawing: dict)` function

File: docs/resources/pumls/engines/tensorflowengine.puml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        - _model: any
+        - _session: any
+        - _preprocessor: TensorFlowPreprocessor
+        - _postprocessor: TensorFlowPostprocessor
+        + __init__(config: TFModelConfig)
+        + verify(): bool
+        + preprocess(frame: np.ndarray): np.ndarray
+        + _predict(preprocessed_frame: np.ndarray): np.ndarray
+        + postprocess(frame: np.ndarray, outputs: dict): np.ndarray
+        + input_size: Tuple[int, int]
```
Comment (Hulongyin) [inline docs/resources/pumls/engines/tensorflowengine.puml]: (property)

File: docs/resources/pumls/filedb.puml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+!includeurl ./style.puml
+
+package "FileDatabase" {
+
+    abstract class FileDatabase<<abstract>> {
+        + {abstract} get_file(filename: str): str
+    }
+
+    class LocalFileDatabase {
```
Comment (Hulongyin) [inline docs/resources/pumls/filedb.puml:11]: Add `_root_dir` attribute and `__init__(root_dir: str)` function

File: docs/resources/pumls/frame.puml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        + {abstract} encrypt()
+        + {abstract} decrypt()
+    }
+
+    class QATFrameCipher {
+        + encrypt()
+        + decrypt()
+    }
+
+    class Frame {
```
Comment (Hulongyin) [inline docs/resources/pumls/frame.puml:17]: Add
`__init__(provider: StreamProvider, pipeline_id: str, sequence: int, raw_frame: np.ndarray)`
`pipeline_id(property): str`
`pipeline_id(new_str: str)(setter): None`
`sequence(property): int`
`timestamp_new_frame(property): float`
`timestamp_new_frame(timestamp: float)(setter): None` `raw(property): np.ndarray`
`raw(raw_frame: np.ndarray)(setter): None`
`timestamp_infer_start(property): float` `timestamp_infer_start(timestamp: float)(setter): float`
functions

File: docs/resources/pumls/frame.puml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        - provider: StreamProvider
+        - pipeline_id: str
+        - sequence: int
+        - raw: np.ndarray
+        - ts_new: float
+        - ts_infer_start: float
+        --
+        + {static} get_sequence(): int
```
Comment (Hulongyin) [inline docs/resources/pumls/frame.puml:36]: This function is `classmethod`, is there `classmethod` in plantUML?

File: docs/resources/pumls/frame.puml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        + to_blob(): Any
```
Comment (Hulongyin) [inline docs/resources/pumls/frame.puml]: return bytes

File: docs/resources/pumls/frame.puml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        + {static} from_blob(blob: Any): Any
```
Comment (Hulongyin) [inline docs/resources/pumls/frame.puml]: (blob: bytes): Frame

File: docs/resources/pumls/infereng.puml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+@startuml InferenceEngine
+
+!includeurl ./style.puml
+
+package "InferenceEngine" {
+
+    class InferenceInfo {
+        - _device: str
+        - _model_id: str
+        - _id: uuid
```
Comment (Hulongyin) [inline docs/resources/pumls/infereng.puml]: str

File: docs/resources/pumls/infereng.puml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        - _queue_topic: str
+        - _input_size: Tuple[int, int]
+        + id: str (property)
```
Comment (Hulongyin) [inline docs/resources/pumls/infereng.puml:15]: Add `id(new_str: str)(setter): None` function

File: docs/resources/pumls/infereng.puml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        + input_size: Tuple[int, int] (property)
+        + device: str (property)
+        + model_id: str (property)
+        + queue_topic: str (property)
```
Comment (Hulongyin) [inline docs/resources/pumls/infereng.puml:19]: Add `queue_topic(new_queue_topic: str): None` function

File: docs/resources/pumls/infereng.puml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+        + __init__(db: RuntimeDatabaseBase): None
+        + search_engine(framework: str, target: str, device: str, model_name: str, model_version: str): Optional[InferenceInfo]
+        + register_engine(infer_info: InferenceInfo, model_info: ModelInfo): None
+        + unregister_engine(infer_info_id: str): None
+    }
+
+    abstract class InferenceEngine<<abstract>> {
+        + {abstract} verify(): bool
+        + {abstract} preprocess(frame: np.ndarray): np.ndarray
+        + predict(frame: np.ndarray): Tuple[Any, float]
```
Comment (Hulongyin) [inline docs/resources/pumls/infereng.puml]: return Tuple[np.ndarray, float]

File: docs/resources/pumls/infereng.puml [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+        + {abstract} postprocess(frame: np.ndarray, outputs: dict): np.ndarray
+        + {abstract} _predict(preprocessed_frame: np.ndarray): Any
```
Comment (Hulongyin) [inline docs/resources/pumls/infereng.puml]: return dict

File: docs/resources/pumls/inferqueue.puml [hunk 15]
```diff
@@ -0,0 +1,38 @@
+@startuml InferqQueue
+
+!includeurl ./style.puml
+
+package "InferqQueue" {
+
+    abstract class InferQueueClientBase<<abstract>> {
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:7]: Add
`__init__()`
`{abstract} infer_queue_available(topic: str): bool`
`{abstract} register_infer_queue(topic: str)`
`{abstract} unregister_infer_queue(topic: str)`
functions

File: docs/resources/pumls/inferqueue.puml [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+        - _buffer_len: int
+        + {abstract} connect(host: str, port: int): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml]: return None

File: docs/resources/pumls/inferqueue.puml [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+        + {abstract} publish_frame(topic: str, frame: Frame): None
+        + {abstract} get_frame(topic: str): Frame
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:13]: return Optional[Frame]

File: docs/resources/pumls/inferqueue.puml [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+        + {abstract} drop(topic: str): int
+        + buffer_len(): int
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml]: (property)

File: docs/resources/pumls/inferqueue.puml [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+        + buffer_len(new_buffer_len: int): void
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml]: (setter) return None

File: docs/resources/pumls/inferqueue.puml [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    class RedisInferQueueClient {
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:22]: Add
`__init__()`
`infer_queue_available(topic: str): bool`
`register_infer_queue(topic: str)`
`unregister_infer_queue(topic: str)`
functions

File: docs/resources/pumls/inferqueue.puml [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+        - _conn: redis.Redis
+        + connect(host: str = "127.0.0.1", port: int = 6379): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml]: return None

File: docs/resources/pumls/inferqueue.puml [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+        + publish_frame(topic: str, frame: Frame): None
+        + get_frame(topic: str): Frame
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:28]: return Optional[Frame]

File: docs/resources/pumls/inferqueue.puml [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+        + drop(topic: str): int
+    }
+
+    class KafkaInferQueueClient {
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:35]: Add
`__init__()`
`{abstract} infer_queue_available(topic: str): bool`
`{abstract} register_infer_queue(topic: str)`
`{abstract} unregister_infer_queue(topic: str)`
functions

File: docs/resources/pumls/inferqueue.puml [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+        - _conn: KafkaProducer
+        + connect(host: str = "127.0.0.1", port: int = 9092): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml]: return None

File: docs/resources/pumls/inferqueue.puml [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+        + publish_frame(topic: str, frame: Frame): None
+        + get_frame(topic: str): Frame
```
Comment (Hulongyin) [inline docs/resources/pumls/inferqueue.puml:41]: return Optional[Frame]

File: docs/resources/pumls/model.puml [hunk 26]
```diff
@@ -0,0 +1,58 @@
+@startuml Model
+
+!includeurl ./style.puml
+
+package "Model" {
+
+    class ModelMetrics {
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml:7]: Add `__init__(accuracy: float, precision: float, recall: float, f1_score: float, loss: float)` and `__iter__(): Iterator[Tuple[str, Any]]` functions

File: docs/resources/pumls/model.puml [hunk 27]
```diff
@@ -0,0 +1,58 @@
...
+        - _accuracy: float
+        - _precision: float
+        - _recall: float
+        - _f1_score: float
+        - _loss: float
+        + accuracy(): float
+        + precision(): float
+        + recall(): float
+        + f1_score(): float
+        + loss(): float
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (property)

File: docs/resources/pumls/model.puml [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    class ModelDetails {
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml:23]: Add `__init__(name: str, version: str, framework: str, target: str, dtype: str)` and `__iter__(): Iterator[Tuple[str, Any]]` functions

File: docs/resources/pumls/model.puml [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+        - _name: str
+        - _version: str
+        - _framework: str
+        - _target: str
+        - _dtype: str
+        + name(): str
+        + version(): str
+        + framework(): str
+        + target(): str
+        + dtype(): str
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (property)

File: docs/resources/pumls/model.puml [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    class ModelInfo {
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml:39]: Add `__init__(details: ModelDetails, uploaded_date: datetime, metrics: ModelMetrics)` and `__iter__(): Iterator[Tuple[str, Any]]` functions

File: docs/resources/pumls/model.puml [hunk 31]
```diff
@@ -0,0 +1,58 @@
...
+        - _id: uuid.UUID
+        - _details: ModelDetails
+        - _uploaded_date: datetime
+        - _metrics: ModelMetrics
+        + id(): uuid.UUID
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (property)

File: docs/resources/pumls/model.puml [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+        + id(new_id: uuid.UUID): void
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (setter) (new_str: str) return None

File: docs/resources/pumls/model.puml [hunk 33]
```diff
@@ -0,0 +1,58 @@
...
+        + details(): ModelDetails
+        + uploaded_date(): datetime
+        + metrics(): ModelMetrics
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (property)

File: docs/resources/pumls/model.puml [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    class Model {
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml:54]: Add `__init__(model_info: ModelInfo, model_binary: bytes)` and `__iter__():Iterator[Tuple[str, Any]]` functions

File: docs/resources/pumls/model.puml [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+        - _model_info: ModelInfo
+        - _model_binary: bytes
+        + model_info(): ModelInfo
+        + model_binary(): bytes
```
Comment (Hulongyin) [inline docs/resources/pumls/model.puml]: (property)

File: docs/resources/pumls/pipeline.puml [hunk 36]
```diff
@@ -0,0 +1,25 @@
+@startuml Pipeline
+
+!includeurl ./style.puml
+
+package "Pipeline" {
+
+class Pipeline {
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml:7]: Add `__init__(provider: StreamProvider, infer_engine_info: InferenceInfo)` and `__iter__(): Iterator[Tuple[str, Dict[str, Any]]]` functions

File: docs/resources/pumls/pipeline.puml [hunk 37]
```diff
@@ -0,0 +1,25 @@
...
+    -_id: uuid.UUID
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml]: str

File: docs/resources/pumls/pipeline.puml [hunk 38]
```diff
@@ -0,0 +1,25 @@
...
+    -_provider: StreamProvider
+    -_info_engine_info: InferenceInfo
+    +id(): uuid.UUID
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml]: (property) return str

File: docs/resources/pumls/pipeline.puml [hunk 39]
```diff
@@ -0,0 +1,25 @@
...
+    +id(new_id: uuid.UUID): void
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml]: (setter) (new_str: str) return None

File: docs/resources/pumls/pipeline.puml [hunk 40]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+class PipelineManager {
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml:18]: Add `__init__(db: RuntimeDatabaseBase)` function

File: docs/resources/pumls/pipeline.puml [hunk 41]
```diff
@@ -0,0 +1,25 @@
...
+    -_db: RuntimeDatabaseBase
+    +register_pipeline(pipeline_obj: Pipeline): void
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml]: return None

File: docs/resources/pumls/pipeline.puml [hunk 42]
```diff
@@ -0,0 +1,25 @@
...
+    +unregister_pipeline(pipeline_id: str): void
```
Comment (Hulongyin) [inline docs/resources/pumls/pipeline.puml]: return None

File: docs/resources/pumls/processors/postprocessor.puml [hunk 43]
```diff
@@ -0,0 +1,26 @@
...
+
+!includeurl ../style.puml
+
+package "PostProcessor" {
+
+    abstract class Postprocessor<<abstract>> {
+        + postprocess(frame: np.ndarray, outputs: dict): np.ndarray
+    }
+
+    class ObjectDetectionPostprocessor {
```
Comment (Hulongyin) [inline docs/resources/pumls/processors/postprocessor.puml:11]: Add `__init__(drawing: dict)` function

File: docs/resources/pumls/processors/postprocessor.puml [hunk 44]
```diff
@@ -0,0 +1,26 @@
...
+        - _drawing: dict
+        + postprocess(frame: np.ndarray, outputs: dict): np.ndarray
+    }
+
+    class FaceRecognitionPostprocessor {
```
Comment (Hulongyin) [inline docs/resources/pumls/processors/postprocessor.puml:18]: Add `__init__(drawing: dict)` function

File: docs/resources/pumls/rtdb.puml [hunk 45]
```diff
@@ -0,0 +1,30 @@
+@startuml RuntimeDatabase
+
+!includeurl ./style.puml
+
+package "RuntimeDatabase" {
+
+    abstract class RuntimeDatabaseBase<<abstract>> {
+        + abstract connect(): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None

File: docs/resources/pumls/rtdb.puml [hunk 46]
```diff
@@ -0,0 +1,30 @@
...
+        + abstract save_table_object_dict(table: str, obj: str, d: dict): void
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None?

File: docs/resources/pumls/rtdb.puml [hunk 47]
```diff
@@ -0,0 +1,30 @@
...
+        + abstract get_table_object_dict(table: str, obj: str): dict
+        + abstract get_all_table_objects_dict(table: str): dict
+        + abstract check_table_object_exist(table: str, obj: str): bool
+        + abstract del_table_object(table: str, obj: str): void
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None?

File: docs/resources/pumls/rtdb.puml [hunk 48]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    class RedisDB {
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml:16]: Add `__init__()` function

File: docs/resources/pumls/rtdb.puml [hunk 49]
```diff
@@ -0,0 +1,30 @@
...
+        - _conn: redis.Redis
+        + connect(host: str, port: int, db: int): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None

File: docs/resources/pumls/rtdb.puml [hunk 50]
```diff
@@ -0,0 +1,30 @@
...
+        + save_table_object_dict(table: str, obj: str, d: dict): void
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None?

File: docs/resources/pumls/rtdb.puml [hunk 51]
```diff
@@ -0,0 +1,30 @@
...
+        + get_table_object_dict(table: str, obj: str): dict
+        + get_all_table_objects_dict(table: str): dict
+        + check_table_object_exist(table: str, obj: str): bool
+        + del_table_object(table: str, obj: str): void
```
Comment (Hulongyin) [inline docs/resources/pumls/rtdb.puml]: return None?

File: docs/resources/pumls/stream.puml [hunk 52]
```diff
@@ -0,0 +1,62 @@
+@startuml Stream
+
+!includeurl ./style.puml
+
+package "Stream" {
+
+    abstract class StreamProvider<<abstract>> {
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml:7]: Add `__init__(name: str, pathname: str)` and `__iter__(): Iterator[Tuple[str, Any]]` functions

File: docs/resources/pumls/stream.puml [hunk 53]
```diff
@@ -0,0 +1,62 @@
...
+        - _pathname: str
+        - _raw_size: Tuple[int, int]
+        - _raw_fps: int
+        - _seq: int
+        - _target_fps: int
+        + name(): str
+        + pathname(): str
+        + raw_size(): Tuple[int, int]
+        + raw_fps(): int
+        + target_fps(): int
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (property)

File: docs/resources/pumls/stream.puml [hunk 54]
```diff
@@ -0,0 +1,62 @@
...
+        + target_fps(new_val): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (setter) (new_val: int) return None

File: docs/resources/pumls/stream.puml [hunk 55]
```diff
@@ -0,0 +1,62 @@
...
+        + {abstract} verify(): bool
+        + {abstract} read_raw_frame(): numpy.ndarray
+        + {abstract} open(): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None

File: docs/resources/pumls/stream.puml [hunk 56]
```diff
@@ -0,0 +1,62 @@
...
+        + {abstract} close(): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None?

File: docs/resources/pumls/stream.puml [hunk 57]
```diff
@@ -0,0 +1,62 @@
...
+
+    }
+
+    class CameraSource {
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml:29]: Add `__init__(name: str, pathname: str="/dev/video0")` function

File: docs/resources/pumls/stream.puml [hunk 58]
```diff
@@ -0,0 +1,62 @@
...
+        - _device_obj: any
+        + dev_num(): int
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: update to `-_dev_num(): int`

File: docs/resources/pumls/stream.puml [hunk 59]
```diff
@@ -0,0 +1,62 @@
...
+        + verify(): bool
+        + read_raw_frame(): numpy.ndarray
+        + open(): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None

File: docs/resources/pumls/stream.puml [hunk 60]
```diff
@@ -0,0 +1,62 @@
...
+        + close(): void
+    }
+
+    class FileSource {
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml:40]: Add `__init__(name: str, pathname: str="classroom.mp4")` function

File: docs/resources/pumls/stream.puml [hunk 61]
```diff
@@ -0,0 +1,62 @@
...
+
+    }
+
+    class CameraSource {
+        - _device_obj: any
+        + dev_num(): int
+        + verify(): bool
+        + read_raw_frame(): numpy.ndarray
+        + open(): bool
+        + close(): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None

File: docs/resources/pumls/stream.puml [hunk 62]
```diff
@@ -0,0 +1,62 @@
...
+
+    class FileSource {
+        - _file_db: FileDatabase
+        - _file_object: any
+        - _file_path: str
+        - _frame_counter: int
+        - _max_frame: int
+        + verify(): bool
+        + open(): bool
+        + close(): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None

File: docs/resources/pumls/stream.puml [hunk 63]
```diff
@@ -0,0 +1,62 @@
...
+        + read_raw_frame(): numpy.ndarray
+        + file_db(): FileDatabase
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (property)

File: docs/resources/pumls/stream.puml [hunk 64]
```diff
@@ -0,0 +1,62 @@
...
+        + file_db(new_val): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (setter) (new_val: FileDatabase) return None

File: docs/resources/pumls/stream.puml [hunk 65]
```diff
@@ -0,0 +1,62 @@
...
+        + target_fps(): int
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (property)

File: docs/resources/pumls/stream.puml [hunk 66]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    class StreamProcessor {
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml:57]: Add `__init__(provider: StreamProvider)` function

File: docs/resources/pumls/stream.puml [hunk 67]
```diff
@@ -0,0 +1,62 @@
...
+        - _provider: StreamProvider
+        + provider: StreamProvider
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: (property)

File: docs/resources/pumls/stream.puml [hunk 68]
```diff
@@ -0,0 +1,62 @@
...
+        + prepare(): void
```
Comment (Hulongyin) [inline docs/resources/pumls/stream.puml]: return None

File: docs/resources/pumls/streambroker.puml [hunk 69]
```diff
@@ -0,0 +1,29 @@
+@startuml StreamBroker
+
+!includeurl ./style.puml
+
+package "StreamBroker" {
+
+    abstract class StreamBrokerClientBase<<abstract>> {
+        + abstract connect(host: str, port: int): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml]: return None

File: docs/resources/pumls/streambroker.puml [hunk 70]
```diff
@@ -0,0 +1,29 @@
...
+        + abstract publish_frame(topic: str, frame: Frame): void
+    }
+
+    class RedisStreamBrokerClient {
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml:12]: Add `__init__()` function

File: docs/resources/pumls/streambroker.puml [hunk 71]
```diff
@@ -0,0 +1,29 @@
...
+        - _conn: redis.Redis
+        + connect(host: str, port: int): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml:16]: return None

File: docs/resources/pumls/streambroker.puml [hunk 72]
```diff
@@ -0,0 +1,29 @@
...
+        + publish_frame(topic: str, frame: Frame): void
+    }
+
+    class KafkaStreamBrokerClient {
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml:20]: Add `__init__()` function

File: docs/resources/pumls/streambroker.puml [hunk 73]
```diff
@@ -0,0 +1,29 @@
...
+        - _conn: kafka.KafkaProducer
+        + connect(host: str, port: int): bool
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml:24]: return None

File: docs/resources/pumls/frame.puml (hunk 7 above)
Comment (leyao-daily) [reply docs/resources/pumls/frame.puml:36]: no, it can be delcared as static.

**Comment by leyao-daily** [comment]
All fixed.

File: docs/resources/pumls/frame.puml [hunk 74]
```diff
@@ -0,0 +1,49 @@
...
+        + timestamp_infer_start(): float (property)
+        + raw(): np.ndarray (property)
+        + pipeline_id(new_str: str) (setter)
+        + timestamp_new_frame(timestamp: float) (setter)
+        + timestamp_infer_start(timestamp: float) (setter)
+        + raw(raw_frame: np.ndarray) (setter)
+        + __init__(provider: StreamProvider, pipeline_id: str, sequence: int, raw: np.ndarray)
+        + {static} get_sequence(): int
+        + to_blob(): bytes
+        + {static} from_blob(blob: Bbytes): Frame
```
Comment (Hulongyin) [inline docs/resources/pumls/frame.puml]: typo: bytes

File: docs/resources/pumls/stream.puml (hunk 58 above)
Comment (Hulongyin) [reply docs/resources/pumls/stream.puml]: `_dev_num()` is a private method now.

File: docs/resources/pumls/streambroker.puml [hunk 75]
```diff
@@ -0,0 +1,33 @@
...
+
+    abstract class StreamBrokerClientBase<<abstract>> {
+        + {abstract} connect(host: str, port: int)
+        + {abstract} publish_frame(topic: str, frame: Frame)
+    }
+
+    class RedisStreamBrokerClient {
+        - _conn: redis.Redis
+        ---
+        + __init__(host: str, port: int)
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml]: The `__init__` function has no parameter now.

File: docs/resources/pumls/streambroker.puml [hunk 76]
```diff
@@ -0,0 +1,33 @@
...
+        + connect(host: str, port: int): bool
+        + publish_frame(topic: str, frame: Frame)
+    }
+
+    class KafkaStreamBrokerClient {
+        - _conn: kafka.KafkaProducer
+        ---
+        + __init__(host: str, port: int)
```
Comment (Hulongyin) [inline docs/resources/pumls/streambroker.puml]: The `__init__` function has no parameter now.

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Pull request merged**